assets/src/edit-story/components/inspector/design/designPanel.js

```diff
@@ -102,8 +102,6 @@ function DesignPanel({
       (evt) => {
         if (evt) {
           evt.preventDefault();
-          // Reset.
-          setElementUpdates({});
```

Comment:
This call was forces a re-render of the component before the updated element values come down the pipeline, thus for one cycle the panel child components are re-rendered with stale data.  This prevented the ability to hit `Enter`, save the data, _and_ select the contents of the input box (because the contents would go from new data, to stale data, back to new data, and that switch would force a deselect).  Removing this reset call, fixed the issue and doesn't seem to cause any other problems as the way `elementUpdates` is updated, previous update data gets overwritten by the new updates.  All the tests pass with this line taken out, and now the `Enter`, save, and select logic works... However, if it's too risky to just remove this reset logic, let me know and I'll try and figure out a work around.

Reply:
One thing I'm noticing currently is that the history seems to create entries for values even if these haven't changed.
For example:
- Add `23horse` to the input
- Now add `23bear` to the input

Even though the original value was really changed only once (from the initial value to `23`), there seem to be 2 history entries for that and getting the original value back needs two "Undo" clicks. Thoughts?

Reply:
Yup you're right Miina, this change does effect the history, it does seem to be adding in extra undo markers when it's not needed.  I'm going to revert this change.  But that means hitting enter and having the contents of the text input selected won't work again.  I'll see if I can figure out a workaround.